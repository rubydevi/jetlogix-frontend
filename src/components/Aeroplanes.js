import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import AuthContext from '../context/AuthProvider';
import { fetchAeroplanes } from '../redux/aeroplanes/aeroplanesActions';
import AeroplaneItem from './AeroplaneItem';

const Aeroplane = () => {
  const dispatch = useDispatch();
  const aeroplanesData = useSelector((state) => state.aeroplanes.aeroplanes);
  const loading = useSelector((state) => state.aeroplanes.loading);
  const error = useSelector((state) => state.aeroplanes.error);

  const { id } = useContext(AuthContext);

  useEffect(() => {
    dispatch(fetchAeroplanes(id));
  }, [dispatch, id]);

  if (loading) {
    return (
      <div className="d-flex justify-content-center align-items-center vh-100">
        <div className="spinner-border" role="status">
          <span className="visually-hidden">Loading...</span>
        </div>
      </div>
    );
  }

  if (error) {
    return (
      <div className="d-flex justify-content-center align-items-center vh-100">
        Error:
        {error}
      </div>
    );
  }

  return (
    <div className="d-flex justify-content-center align-items-center vh-100">
      <div>
        <h1 className="d-flex justify-content-center mb-5">
          Private Airplanes
        </h1>

        {aeroplanesData.aeroplanes && aeroplanesData.aeroplanes.length > 0 ? (
          <ul className="list-unstyled d-flex flex-wrap">
            {aeroplanesData.aeroplanes.map((aeroplane) => (
              <AeroplaneItem
                key={aeroplane.id}
                aeroplane={aeroplane}
                classNames={{
                  button: 'card m-2',
                  aeroplaneBody: 'card-body',
                  image: 'card-img-top',
                  title: 'card-title fw-bold text-uppercase',
                  description: 'card-text',
                }}
              />
            ))}
          </ul>
        ) : (
          <p>No aeroplanes available.</p>
        )}
      </div>
    </div>
  );
};

export default Aeroplane;
