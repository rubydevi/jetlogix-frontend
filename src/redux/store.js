import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservations: reservationsReducer,
  },
});

export default store;
