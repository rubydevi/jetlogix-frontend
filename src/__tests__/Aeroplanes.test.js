import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Aeroplanes from '../components/Aeroplanes';

test('renders Aeroplanes component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Aeroplanes />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
