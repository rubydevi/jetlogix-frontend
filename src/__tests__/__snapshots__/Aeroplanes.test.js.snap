// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Aeroplanes component correctly 1`] = `
<div>
  <div
    class="d-flex justify-content-center align-items-center vh-100"
    role="status"
    style=""
  >
    <li
      class="list-group-item"
      style="width: 17rem;"
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <img
            alt=""
            class=" skeleton  skeleton-card-img"
          />
        </div>
        <div
          class="d-flex justify-content-center align-items-center  w-75"
        >
          <div
            class="h6 skeleton  skeleton-card-header"
          />
        </div>
        <p
          class=" skeleton skeleton-card-text"
        />
        <p
          class=" skeleton  skeleton-card-text"
        />
        <p
          class=" skeleton  skeleton-card-text"
        />
        <p
          class=" skeleton  skeleton-card-text"
        />
        <p
          class=" skeleton  skeleton-card-text"
        />
      </div>
    </li>
    <div
      class=" d-none d-lg-block d-lg-flex justify-content-center align-items-center "
    >
      <li
        class="list-group-item"
        style="width: 17rem;"
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <img
              alt=""
              class=" skeleton  skeleton-card-img"
            />
          </div>
          <div
            class="d-flex justify-content-center align-items-center  w-75"
          >
            <div
              class="h6 skeleton  skeleton-card-header"
            />
          </div>
          <p
            class=" skeleton skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
        </div>
      </li>
      <li
        class="list-group-item"
        style="width: 17rem;"
      >
        <div
          class=""
        >
          <div
            class=""
          >
            <img
              alt=""
              class=" skeleton  skeleton-card-img"
            />
          </div>
          <div
            class="d-flex justify-content-center align-items-center  w-75"
          >
            <div
              class="h6 skeleton  skeleton-card-header"
            />
          </div>
          <p
            class=" skeleton skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
          <p
            class=" skeleton  skeleton-card-text"
          />
        </div>
      </li>
    </div>
  </div>
</div>
`;
